File: Arrays/PairSums/pairSum.js
(function () {
  let arr = [1, 3, 6, 2, 5, 4, 3, 2, 4];
  let x = 7;
  let ans = 0;

  for (let i = 0; i < arr.length; i++) {
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[i] + arr[j] == x) {
        ans++;
      }
    }
  }

  console.log(ans);
})();

File: Arrays/PairSums/pairSum.py
from sys import stdin
import sys

def pairSum(arr, n, x) :
    ans = 0

    for i in range(n) :
        for j in range(i+1, n) :
            if arr[i] + arr[j] == x :
                ans += 1
    
    return ans


def takeInput() :
    n = int(stdin.readline().strip())
    if n == 0 :
        return list(), 0

    arr = list(map(int, stdin.readline().strip().split(" ")))
    return arr, n

t = int(stdin.readline().strip())

while t > 0 :
    
    arr, n = takeInput()
    x = int(stdin.readline().strip())
    print(pairSum(arr, n, x))

    t -= 1